### DISCUSSION TITLE
Eliminate the warning when starting nvim

### DISCUSSION THREAD

**Discussion by UbeCc** [max397574/better-escape.nvim#83] [Q&A] [answered]
Hi! I've update better-escape.nvim and got the warning when entering nvim

```text
[better-escape.nvim]: Rewrite! Check: https://github.com/max397574/better-escape.nvim
```
I don't know how to fixed it. Can someone give me some suggestions, thanks! TAT

**Comment by max397574** [comment]
can you send your config?
you get this warning when you use the old way of configuring better-escape

**Reply by UbeCc** [reply]
> can you send your config? you get this warning when you use the old way of configuring better-escape

Thanks! But sorry that I'm new to nvim, so I grepped whatever related to `better-escape` plugin at `~/.config` and got:

In `~/.config/lazy-lock.json`,

```lua
"indent-blankline.nvim": { "branch": "master", "commit": "9637670896b68805430e2f72cf5d16be5b97a22a" },
```

In `~/.configu/lua/modules/plugins/editor.lua`,

```lua
editor["max397574/better-escape.nvim"] = {
	lazy = true,
	event = { "CursorHold", "CursorHoldI" },
	config = require("editor.better-escape"),
}
```

And are there any other configurations may related to `better-escape`?

**Reply by max397574** [reply]
whatever is inside `lua/editor/better-escape` is what matters here

**Reply by UbeCc** [reply]
> whatever is inside `lua/editor/better-escape` is what matters here

🤔 I've downloaded nvim from (nvimdots)[https://github.com/ayamir/nvimdots/].
And I didn't see the path `lua/editor/better-escape`

**Reply by max397574** [reply]
well somewhere you surely have a file where you setup better-escape
if you can't find your own config for the plugin I also can't help you fix it

You either have the `keys`, `clear_empty_lines` or `mapping` field set in your config

**Comment by UbeCc** [comment]
Sincerely thanks for your advice.
I searched my `.local` folder and find my `nvim` maganes packages through `lazy`. And `lazy` will put its configuration into `~/.local/share/nvim/site/lazy`. I find the `notification` at `~/.local/share/nvim/site/lazy/better-escape.nvim/lua/better_escape.lua`

```
  if settings.keys or settings.clear_empty_lines then
      vim.notify(
          "[better-escape.nvim]: Rewrite! Check: https://github.com/max397574/better-escape.nvim",
          vim.log.levels.WARN,
          {}
      )
  end
  if settings.mapping then
      vim.notify(
          "[better-escape.nvim]: Rewrite! Check: https://github.com/max397574/better-escape.nvim",
          vim.log.levels.WARN,
          {}
      )
      if type(settings.mapping) == "string" then
          settings.mapping = { settings.mapping }
      end
      for _, mapping in ipairs(settings.mappings) do
          settings.mappings.i[mapping:sub(1, 2)] = {}
          settings.mappings.i[mapping:sub(1, 1)][mapping:sub(2, 2)] =
              settings.keys
      end
  end
```

After I removed the notification in the file, it works. Thanks!

**Reply by max397574** [reply]
that's wrong
here you are actually modifying the plugin source code

so you're not actually fixing things
this will cause issues when updating and when you update the plugin this code will just be readded

somewhere in your config you definitely have a config for better-escape which is invalid

do you have your whole config somewhere on github?

**Comment by UbeCc** [comment]
It's my fault ...
I forgot to add `*` when I use `find` to find `better-escape.lua`, and it is at lua/modules/configs/editor/better-escape.lua

```
    8 return function()
    7 │···require("modules.utils").load_plugin("better_escape", {
    6 │···│···mapping = { "jk", "jj" }, -- a table with mappings to use
    5 │···│···timeout = vim.o.timeoutlen, -- the time in which the keys must be hit in ms. Use option timeoutlen by default
    4 │···│···clear_empty_lines = false, -- clear line after escaping if there is only whitespace
    3 │···│···keys = "<Esc>", -- keys used for escaping, if it is a function will use the result everytime
    2 │···│···-- example(recommended)
    1 │···│···-- keys = function()
  9   │···│···--   return vim.api.nvim_win_get_cursor(0)[2] > 1 and '<Esc>l' or '<Esc>'  github-actions[bot], 9 months ago - chore(lockfile): auto update lazy-lock.json
    1 │···│···-- end,
    2 │···})
    3 end
```

**Reply by max397574** [reply, marked as answer]
yes there is the issue
you could basically just remove everything between the {}

because these are all like the new defaults

**Reply by UbeCc** [reply]
Thanks for you help!
